### PR TITLE
"No repeated property names" error in object literals is duplicated in strict mode

### LINKED ISSUE #46815: "No repeated property names" error in object literals is duplicated in strict mode
In strict mode (`strict: true`, inside a class or inside a module), duplicated property names get two errors instead of one:

```ts
// @strict: false
const first = { a: 1, a: 2 }
class C {
    m() {
        const second = { a: 1, a: 2 }
        return second.a
    }
}
```

Expected: one error on each `a: 2`.
Actual: two errors on the `a: 2` inside the class:

1. Diagnostics.An_object_literal_cannot_have_multiple_properties_with_the_same_name_in_strict_mode
2. Diagnostics.Duplicate_identifier_0

In fact, the first error is better, but isn't used for `first`;

Expected: On `first`s `a: 2`, the error should be "An object literal cannot have multiple properties with the same name."
Actual: Duplicate identifier 'a'.

This inconsistency arose because Typescript originally targetted ES3 and ES5; duplicate property names are only an error in ES5's strict mode. In fact, they're not even an error in ES2015+'s strict mode. But this is a good error, and it should apply everywhere.

Expected: The error should be "An object literal cannot have multiple properties with the same name."
Actual: "An object literal cannot have multiple properties with the same name in strict mode."

Thanks to https://stackoverflow.com/questions/36321790/js-two-or-more-object-properties-with-the-same-name-in-non-strict-mode for pointing out the spec changes.

### PR DESCRIPTION
**Pull request by Yuuki77**
improved the error messages when object property name is duplicated.
smaple
```typescript
// @strict: false
const first = { a: 1, a: 2 }
class C {
    m() {
        const second = { a: 1, a: 2 }
        return second.a
    }
}
````
before

```
test.ts:3:23 - error TS2300: Duplicate identifier 'a'.

3 const first = { a: 1, a: 2 };
                        ~

test.ts:6:32 - error TS1117: An object literal cannot have multiple properties with the same name in strict mode.

6         const second = { a: 1, a: 2 };
                                 ~

test.ts:6:32 - error TS2300: Duplicate identifier 'a'.

6         const second = { a: 1, a: 2 };
                                 ~

[12:48:19 PM] Found 3 errors. Watching for file changes
```
now
```
test.ts:3:23 - error TS1117: An object literal cannot have multiple properties with the same name.

3 const first = { a: 1, a: 2 };
                        ~

test.ts:6:32 - error TS1117: An object literal cannot have multiple properties with the same name.

6         const second = { a: 1, a: 2 };
```

this is my first pr to this project.
Let me know if there is anything I am missing 😄 

Fixes #46815

### CODE REVIEW

**Comment by Yuuki77** [comment]
master is failing as well 🤔

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -43128,8 +43128,11 @@ namespace ts {
                         seen.set(effectiveName, currentKind);
                     }
                     else {
-                        if ((currentKind & DeclarationMeaning.PropertyAssignmentOrMethod) && (existingKind & DeclarationMeaning.PropertyAssignmentOrMethod)) {
-                            grammarErrorOnNode(name, Diagnostics.Duplicate_identifier_0, getTextOfNode(name));
+                        if ((currentKind & DeclarationMeaning.Method) && (existingKind & DeclarationMeaning.Method)) {
+                                grammarErrorOnNode(name, Diagnostics.Duplicate_identifier_0, getTextOfNode(name));
+                        }
+                        else if ((currentKind & DeclarationMeaning.PropertyAssignment) && (existingKind & DeclarationMeaning.PropertyAssignment)) {
+                                grammarErrorOnNode(name, Diagnostics.An_object_literal_cannot_have_multiple_properties_with_the_same_name, getTextOfNode(name));
```
Comment (sandersn) [inline src/compiler/checker.ts:43135]: indentation should be 4 spaces, not 8

**Review by sandersn — CHANGES_REQUESTED** [review]
Looks good, just need to fix indentation. (And I hope the build is no longer broken, you might have to merge from main to get that.)

**Comment by Yuuki77** [comment]
@sandersn thank you for the reviews!
I have updated it.
Could you take a look when you have time?

**Review by sandersn — APPROVED** [review]

**Pull request merged**